src/Node/Crypto/Types.purs

```diff
@@ -0,0 +1,276 @@
+module Node.Crypto.Types
+  ( Algorithm(..)
+  , AuthAlgorithm(..)
+  , Password(..)
+  , Plaintext(..)
+  , Ciphertext(..)
+  , Key(..)
+  , AuthTag(..)
+  , InitializationVector(..)
+  ) where
+
+-- Notes:
+-- iv is optional (null if no iv needed)
+-- setAuthTag is optional: it is needed for authenticated encryption modes (GCM, CCM and OCB are currently supported)
+-- key and iv should be utf-8 encoded or buffers
+-- createCipher authTagLength needed for CCM or OCB, but is not added, yet
+
+import Data.Show (class Show, show)
+import Data.Newtype (class Newtype)
+import Node.Buffer (Buffer)
+import Data.Eq (class Eq)
+
+data AuthAlgorithm
+  = AES256GCM
+
+data Algorithm
+  = AES128
+  | AES192
+  | AES256
+  | AES256CBC
+  | WithAuth AuthAlgorithm
+
+newtype Password
+  = Password String
+
+instance ntPassword :: Newtype Password String
+
+newtype Key
+  = Key String
+
+instance ntKey :: Newtype Key String
+
+newtype Plaintext
+  = Plaintext String
+
+instance ntPlaintext :: Newtype Plaintext String
+derive newtype instance eqPlaintext :: Eq Plaintext
+
+instance showPlaintext :: Show Plaintext where
+  show (Plaintext p) = p
+
+newtype Ciphertext
+  = Ciphertext String
+
+instance ntCiphertext :: Newtype Ciphertext String
+derive newtype instance eqCiphertext :: Eq Ciphertext
+
+instance showCiphertext :: Show Ciphertext where
+  show (Ciphertext c) = c
+
+newtype AuthTag
+  = AuthTag Buffer
+
+instance ntAuthTag :: Newtype AuthTag Buffer
+
+newtype InitializationVector
+  = InitializationVector String
```

Comment:
`Some types can be used as `iv`.
At least newtype is not suitable.

I think better that you want to use `data` or typeclass and instantiate `String`.